exporter/metric/metric.go

```diff
@@ -272,25 +291,92 @@ func (me *metricExporter) recordToMdpb(record *export.Record) *googlemetricpb.Me
 	}
 }
 
+// checkPrefixInMapKeys checks whether the non-empty prefix string
+//  is in the values of the map
+func checkPrefixInMapKeys(labelMap map[string]string, prefix string) bool {
+	if len(prefix) == 0 {
+		return false
+	}
+	for key := range labelMap {
+		if strings.HasPrefix(key, prefix) {
+			return true
+		}
+	}
+	return false
+}
+
 // resourceToMonitoredResourcepb converts resource in OTel to MonitoredResource
 // proto type for Cloud Monitoring.
 //
 // https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.monitoredResourceDescriptors
-func (me *metricExporter) resourceToMonitoredResourcepb(_ *resource.Resource) *monitoredrespb.MonitoredResource {
-	// TODO: Implement the process to convert Resource to MonitoringResoruce.
-	// ref. https://cloud.google.com/monitoring/api/resources
-	// This method only returns global resource for now.
-
-	// "global" only accepts "project_id" for label.
-	// https://cloud.google.com/monitoring/api/resources#tag_global
+func (me *metricExporter) resourceToMonitoredResourcepb(res *resource.Resource) *monitoredrespb.MonitoredResource {
+
+	if res == nil || res.Len() == 0 {
```

Comment:
Since you end up doing the same initialization in both branches of code, you could avoid a bit of duplication by doing:

```go
monitoredRes := &monitoredrespb.MonitoredResource{
	Type: "global",
	Labels: map[string]string{
		"project_id": me.o.ProjectID,
	},
}

if res == nil || res.Len() == 0 {
	return monitoredRes 
}

...
monitoredRes.Labels = transformResource(match, resLabelMap)
```